### ISSUE TITLE
Default Value for Latitude and Longitude, empty TIV col should not be required

### ISSUE THREAD

**Issue by sstruzik** [OasisLMF/ODS_OpenExposureData#132]
## Description

Default value for Latitude and Longitude are 0 which is actually a valid position and may therefore misplace the location.
This value needs to stay null.

For convenience, It would be better if TIV col were not required for empty values.

## Scope of change 
 
- [x] Location File 
- [ ] Accounts File 
- [ ] Reinsurance Scope 
- [ ] Reinsurance Info 

## Impact of change
 
 
This change may correct the misplacement of certain location.

## Data type, default values, are blanks allowed, list valid values
 
sheet OED Input Fields:

| **Field Name** | **Required Field** | **Allow blanks?** | **Default values** |
|----------------|---------------|
| Latitude | O | YES |       n/a        | 
| Longitude | O | YES |      n/a         | 
| BuildingTIV | R | YES | 0 |
| ContentsTIV | R | YES | 0 |
| BITIV | R | YES | 0 |

**Comment by MattDonovan82** [comment]
this is a small change in the spec but will be a minor update so v3.1

We will leave this here to let the community comment, but don't foresee any issues.

**Comment by fl-ndaq** [comment]
@MattDonovan82 you suggest a 3.1 update for this but the actual release is 3.0.2.
This is not a small change if we consider that modellers could have to change  their keys_lookup logic to deal with it

**Comment by jones-matt** [comment]
I don't see why this change is needed. 0,0 isn't a position that any modelled assets will ever use and it will likely require extra work from most modellers.

**Comment by benhayes21** [comment]
The problem is where blank values are replaced with the default in OED - (0,0) for lat/lon - which is causing the 0,0 to be used in the lat/lon lookup rather than, say, postcode

**Comment by aiste-kalinauskaite** [comment]
This is not advisable. Lat/long (0,0) is in the middle of the ocean, which means for any property modelling is irrelevant. I am unaware that there is an offshore platform in such coordinates either. Mixing data types for this or any other column is not a good idea, as it complicates data preparation for all users (most users don't store data in csvs, it sits somewhere else). This is one of the primary reasons why we have default float values for lat/long. As already mentioned, this change is a big change for all model vendors too who would need to adjust key lookups.

**Comment by JBADougBurns** [comment]
Maybe I'm missing something about the bigger picture here. I'm not sure why making latitude and longitude values optional results in such a big impact? Previously to oasislmf 1.27 (where the OED lat-lon defaults were not enforced) location files could happily contain no coordinate information. Would most model users not already have some portfolio preparation method with respect to previous versions of oasislmf ready? (either by adding (0, 0) for exposure that don't have a precise location, or by not including that information). By forcing (0, 0) it is a change to our keys lookup (and other model vendors?) so we would prefer the handling to be consistent with older versions of oasislmf.

Are there particular data stores that are often used and do not allow null values? To my knowledge oasislmf can read portfolio files in csv and parquet formats which can handle null values OK. I maybe missing other data formats?

**Comment by benhayes21** [comment]
I think that this change is actually reverting back to "expected" behaviour that we saw in pre-1.27 oasislmf. It should actually prevent model providers' keys services from breaking rather than causing them to break.

Prior to 1.27, the oasis package would not do anything with the OED data and it would be passed through to the keys service without change (except for any per-analysis adjustment process). So if a blank lat-lon was presented, that is what the keys service would see and it could deal with it however it was programmed to do so.

From 1.27, we have introduced OED validation into the oasis package, a part of which involved replacing blank values with the default from the OED spec for those fields affected. So in the case of blank lat-lons being presented, the default of (0,0) was being substituted in by the package, which the keys service then received and dealt with. This was causing problems for existing model keys services which look for a lat-lon to place the location prior to looking for other options (postcode, etc.) as they are reading the new (0,0) value as a valid lat-lon and rejecting the location as out of scope.

Making the default (n/a,n/a) means that blanks will flow through from OED to the keys service.
